### PR TITLE
Fix ambiguities in spec inertial documentation

### PR DESCRIPTION
**Pull request by scpeters**
# 🎉 New feature

Part of https://github.com/ignitionrobotics/sdf_tutorials/issues/70

## Summary

As noted in https://github.com/ignitionrobotics/sdf_tutorials/issues/70, there are some ambiguities in the definition of inertial properties in the SDFormat specification, particularly the pose and inertia tensor terms. Thanks to @mitiguy, @sherm1, and @EricCousineau-TRI for providing the text of these changes.

These changes should be propagated to earlier versions of the spec, both in the other folders of the `sdf12` branch and to earlier supported branches such as `sdf9` and `sdf6`. It also needs to be tested for its rendering on sdformat.org on the following page:

* http://sdformat.org/spec?ver=1.9&elem=link#link_inertial

I have marked this as a draft pull request for now. Once the text has been approved, I will propagate the changes to the other files as needed.

## Test it

* Read the documentation
* TODO: test its rendering on sdformat.org

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [X] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

https://reviewable.io/reviews/ignitionrobotics/sdformat/990#-

---
This change is [Reviewable](https://reviewable.io/reviews/ignitionrobotics/sdformat/990)

### CODE REVIEW

File: sdf/1.9/inertial.sdf [hunk 1]
```diff
@@ -27,24 +38,54 @@
...
     <element name="ixx" type="double" default="1.0" required="1">
-      <description></description>
+      <description>
+        The link’s moment of inertia about Co (the link’s center of mass) for Ĉx.
+      </description>
     </element>
     <element name="ixy" type="double" default="0.0" required="1">
-      <description></description>
+      <description>
+        The link’s product of inertia about Co (the link’s center of mass) for
```
Comment (EricCousineau-TRI) [inline sdf/1.9/inertial.sdf:61]: nit (here and elsewhere) `link’s` => `link's` (different character?)

Comment (scpeters) [reply sdf/1.9/inertial.sdf]: https://github.com/ignitionrobotics/sdformat/pull/990/commits/77630ddd7248f55178cc92acda9cbc0391eaf9d9

File: sdf/1.9/inertial.sdf [hunk 2]
```diff
@@ -27,24 +38,54 @@
...
+      This link's moments of inertia ixx, iyy, izz and products of inertia
+      ixy, ixz, iyz about Co (the link's center of mass) for the unit vectors
+      Ĉx, Ĉy, Ĉᴢ fixed in the center-of-mass-frame C.
+      Note: the orientation of Ĉx, Ĉy, Ĉᴢ relative to L̂x, L̂y, L̂ᴢ is specified
+      by the `pose` tag.
+      To avoid compatibility issues associated with the negative sign
+      convention for product of inertia, align Ĉx, Ĉy, Ĉᴢ with principal
+      inertia directions so that all the products of inertia are zero.
+      For more information about this sign convention, see the following
+      Mathworks documentation for working with CAD tools:
```
Comment (traversaro) [inline sdf/1.9/inertial.sdf:51]: Nitpick: The company is tipically indicated with the W capitalized, see https://www.mathworks.com/ .
```suggestion
      MathWorks documentation for working with CAD tools:
```

**Review by EricCousineau-TRI — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ignitionrobotics/sdformat/990)* status: all files reviewed, 1 unresolved discussion (waiting on @scpeters)

**Review by EricCousineau-TRI — APPROVED** [review]

**Review by scpeters — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ignitionrobotics/sdformat/990)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @EricCousineau-TRI and @traversaro)

___
*[sdf/1.9/inertial.sdf, line 51 at r2](https://reviewable.io/reviews/ignitionrobotics/sdformat/990#-N-zPxIOok40b8siAWat-r2-51:-N02q-117i8hm8B0daxY:bpo9r2r) ([raw file](https://github.com/ignitionrobotics/sdformat/blob/a812c25dafc6783c4f847f897f8e107a619badb8/sdf/1.9/inertial.sdf#L51)):*
<details><summary><i>Previously, traversaro (Silvio Traversaro) wrote…</i></summary><blockquote>

Nitpick: The company is tipically indicated with the W capitalized, see https://www.mathworks.com/ .
```suggestion
      MathWorks documentation for working with CAD tools:
```
</blockquote></details>

Ok, thanks! I applied your suggestion in 1c5df36

**Review by mitiguy — COMMENTED** [review]
This looks great.  Thank you Silvio, Eric, Steve, Addisu, and all.  Much appreciated.

> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/ignitionrobotics/sdformat/990#-:-N07mwKLBJilIR5t0vrD:bapsssw)* status: all files reviewed (commit messages unreviewed), 1 unresolved discussion (waiting on @traversaro)

**Comment by scpeters** [comment]
> It also needs to be tested for its rendering on sdformat.org on the following page:
> 
> * http://sdformat.org/spec?ver=1.9&elem=link#link_inertial

I just tested it, and I think there are some significant issues with URLs based on how we are rendering the content in https://sdformat.org/spec

They can be seen in the current form by interacting with the `//link/pose` documentation:

* http://sdformat.org/spec?ver=1.9&elem=link#link_pose

That documentation contains a link to http://sdformat.org/tutorials?tut=specify_pose but it is not clickable in my tests. None of the text is even selectable. Attempting to select text will drag the whole block, which drops a link to that portion of the page if you drop it into a text document.

So the text of these new URLs is visible but not clickable or selectable. This should be remedied on the web app code.

**Comment by scpeters** [comment]
quick side-by-side comparison with this branch on the left and the existing display on the right

sdf_inertial_rendering

**Comment by EricCousineau-TRI** [comment]
I think even w/ rendering fix, then docs look better.
Perhaps this could merge, then we fix the rendering as a separate task?

**Comment by scpeters** [comment]
> I think even w/ rendering ~fix~ bug, the docs look better.
> Perhaps this could merge, then we fix the rendering as a separate task?

Sure, I'll work on replicating the changes to the other spec folders

**Comment by scpeters** [comment]
> Sure, I'll work on replicating the changes to the other spec folders

I've copied the changes back to spec version 1.4 in https://github.com/ignitionrobotics/sdformat/pull/990/commits/43f2f84ad52ed20916701b4d0ddf9e9a4b040bba and https://github.com/ignitionrobotics/sdformat/pull/990/commits/fc0ba0a5c2c309f240e5455265acd11789a1eab7. I adjusted the wording slightly in spec versions 1.5, 1.6, and 1.7 to account for the presence of `//inertial/pose/@frame` and `//inertial/pose/@relative_to` attributes in those versions of the spec that were removed in spec version 1.8.

File: sdf/1.4/inertial.sdf [hunk 3]
```diff
@@ -1,34 +1,77 @@
...
+    <description>
+      This pose (translation, rotation) describes the position and orientation
+      of the link's center-of-mass-frame C relative to the link-frame L.
+      The first three components (x y z) specify the position vector from Lo
+      (the link-frame origin) to Co (the link's center of mass) as
+      `x L̂x + y L̂y + z L̂ᴢ`, where L̂x, L̂y, L̂ᴢ are link-frame L's orthogonal unit
+      vectors. The subsequent values characterize C's orientation relative to
+      link-frame L as a sequence of Euler rotations
+      (r p y) documented in http://sdformat.org/tutorials?tut=specify_pose,
+      or as a quaternion (x y z w), where w is the scalar component.
```
Comment (azeey) [inline sdf/1.4/inertial.sdf:22]: We only support quaternions in >= SDFormat 1.9

File: sdf/1.8/inertial.sdf [hunk 4]
```diff
@@ -1,34 +1,91 @@
...
+      The first three components (x y z) specify the position vector from Lo
+      (the link-frame origin) to Co (the link's center of mass) as
+      `x L̂x + y L̂y + z L̂ᴢ`, where L̂x, L̂y, L̂ᴢ are link-frame L's orthogonal unit
+      vectors. The subsequent values characterize C's orientation relative to
+      link-frame L as a sequence of Euler rotations
+      (r p y) documented in http://sdformat.org/tutorials?tut=specify_pose,
+      or as a quaternion (x y z w), where w is the scalar component.
+    </description>
+
+    <attribute name="rotation_format" type="string" default="euler_rpy" required="0">
```
Comment (azeey) [inline sdf/1.8/inertial.sdf:25]: The attributes `rotation_format` and `degree` were added to 1.9.  Was it intentional to add them in `sdf/1.8/inerital.sdf`?

**Review by azeey — CHANGES_REQUESTED** [review]

File: sdf/1.4/inertial.sdf (hunk 3 above)
Comment (sherm1) [reply sdf/1.4/inertial.sdf:22]: Please tell me the quaternion is given as (w x y z)!

Comment (azeey) [reply sdf/1.4/inertial.sdf:22]: To be consistent with ROS and Ignition/Gazebo, we went with xyzw https://github.com/ignitionrobotics/sdf_tutorials/pull/56#issuecomment-919274182

Comment (scpeters) [reply sdf/1.4/inertial.sdf]: fixed in https://github.com/ignitionrobotics/sdformat/pull/990/commits/2a62ad1f2e3df1bd66a6a9b5c90688f589ab493d

File: sdf/1.8/inertial.sdf (hunk 4 above)
Comment (scpeters) [reply sdf/1.8/inertial.sdf]: nope, not intentional. removed in https://github.com/ignitionrobotics/sdformat/pull/990/commits/2a62ad1f2e3df1bd66a6a9b5c90688f589ab493d

**Review by azeey — APPROVED** [review]

File: sdf/1.4/inertial.sdf (hunk 3 above)
Comment (sherm1) [reply sdf/1.4/inertial.sdf]: :( but consistency is a good argument. Eigen stores it that way also but was apparently so ashamed that they hid it by having their Quaternion constructor take w,x,y,z!

Comment (scpeters) [reply sdf/1.4/inertial.sdf]: I hear your pain, but at least the convention is specified in the `@rotation_format` attribute as `quat_xyzw`.

**Pull request merged**